From 3a7d89cbc3605f79d88225b2b3f29a5978a118b4 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Thu, 7 Feb 2019 15:22:58 +0100
Subject: [PATCH] mtd: spi_flash: do not call EN4B for 4B address set

this command is used to switch the normal command set to use 4 byte
adresses. This is not needed when we use 4B command set.

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 drivers/mtd/spi/spi_flash.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/mtd/spi/spi_flash.c b/drivers/mtd/spi/spi_flash.c
index de35707948..8c9f5b5fd5 100644
--- a/drivers/mtd/spi/spi_flash.c
+++ b/drivers/mtd/spi/spi_flash.c
@@ -1135,6 +1135,7 @@ int spi_flash_scan(struct spi_flash *flash)
 
 #ifdef CONFIG_SPI_FLASH_4BYTES_ADDR
 		if (flash->size > SPI_FLASH_16MB_BOUN) {
+			printf("SF : select 4B addr cmd set\n");
 			flash->read_cmd = CMD_READ_ARRAY_FAST_4B;
 			flash->write_cmd = CMD_PAGE_PROGRAM_4B;
 
@@ -1142,8 +1143,6 @@ int spi_flash_scan(struct spi_flash *flash)
 				flash->erase_cmd = CMD_ERASE_4K_4B;
 			else
 				flash->erase_cmd = CMD_ERASE_64K_4B;
-
-			enter_4bytes_addr(flash);
 		}
 #endif
 
